mdc_weight_mgmt/models/product_product.py

```diff
@@ -0,0 +1,25 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    def _get_default_category_id(self):
+        context = self.env.context
+        if context.get('product_categ_default_id'):
+            return self.env.company.product_categ_default_id
+        return super(ProductProduct,self)._get_default_category_id()
+
+    weight_dec_qty = fields.Float(
+        string='Declared Weight',
+        required=True
+    )
+
+    @api.constrains("weight_dec_qty")
+    def _check_weight_dec_qty(self):
+        for product in self:
+            if product.weight_dec_qty <= 0:
+                raise ValidationError(_("The declared weight must be greater than 0."))
```

Comment:
Suggested code (prevent `for` + `if` usage), not tested:
```suggestion
        products_ko_weight_dec_qty = self.filtered(lambda x: x.weight_dec_qty <= 0)
        if products_ko_weight_dec_qty:
            raise ValidationError(_("The declared weight must be greater than 0."))
```